Admin login error results in panic

```
mim login devadmin
 SUCCESS  Updated config file
login starting...
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/mimiro-io/datahub-cli/internal/web.DoAdminLogin(0xc00034a080)
	/Users/andebor/mimiro/datahub-cli/internal/web/token.go:194 +0x42e
github.com/mimiro-io/datahub-cli/internal/login.UseLogin({0x7ffeefbff0ea, 0x8})
	/Users/andebor/mimiro/datahub-cli/internal/login/use.go:76 +0x12a
github.com/mimiro-io/datahub-cli/internal/command.glob..func7(0x26bbbc0?, {0xc0002bdf60, 0x1, 0x1?})
	/Users/andebor/mimiro/datahub-cli/internal/command/ll.go:49 +0xf0
github.com/spf13/cobra.(*Command).execute(0x26bbbc0, {0xc0002bdf30, 0x1, 0x1})
	/Users/andebor/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x26ba7c0)
	/Users/andebor/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x39c
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/andebor/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/mimiro-io/datahub-cli.Execute()
	/Users/andebor/mimiro/datahub-cli/root.go:51 +0x25
main.main()
	/Users/andebor/mimiro/datahub-cli/cmd/cli/main.go:23 +0x17
```

--- a/internal/login/use.go
+++ b/internal/login/use.go
@@ -104,7 +104,7 @@ func UseLogin(alias string) (*config.SignedToken, error) {
 	}
 
 	if err2 != nil {
-		return nil, err
+		return nil, err2
 	}
 
 	data.Type = loginType         // this will upgrade existing ones as they are used

--- a/internal/web/token.go
+++ b/internal/web/token.go
@@ -187,6 +187,12 @@ func DoAdminLogin(cfg *config.Config) (*config.SignedToken, error) {
 	if err != nil {
 		return nil, err
 	}
+	defer res.Body.Close()
+	statusOK := res.StatusCode >= 200 && res.StatusCode < 300
+	if !statusOK {
+		bodyBytes, _ := ioutil.ReadAll(res.Body)
+		return nil, fmt.Errorf("%d: %s", res.StatusCode, string(bodyBytes))
+	}
 
 	decoder := json.NewDecoder(res.Body)
 	response := make(map[string]interface{})

